docs/source/developers/index.rst

```diff
@@ -0,0 +1,43 @@
+.. _developers_guide:
+
+Developer's Guide
+==================
+
+This section contains information for developers who want to contribute to the cuML project.
```

Comment:
I think this is in principle the right place, but we are currently maintaining the majority of our development guidance within the [wiki](https://github.com/rapidsai/cuml/tree/branch-25.04/wiki). I think we should add these instructions first there, clean everything up, and then migrate the instructions to the docs.

Reply:
Aha! That explains why there were no developer docs! Where can I see the wiki? https://github.com/rapidsai/cuml/wiki asks me to create the first page?!

Reply:
Moved it